examples/common/chip-app-server/server.cpp

```diff
@@ -0,0 +1,96 @@
+#include "server.h"
+
+#include <string.h>
+
+#include <inet/IPAddress.h>
+#include <inet/InetError.h>
+#include <inet/InetLayer.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <support/CodeUtils.h>
+#include <support/ErrorStr.h>
+#include <support/logging/CHIPLogging.h>
+#include <sys/param.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/UDP.h>
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::Transport;
+using namespace ::chip::DeviceLayer;
+
+#ifndef EXAMPLE_SERVER_NODEID
+// "nRF5"
+#define EXAMPLE_SERVER_NODEID 12344321
+#endif // EXAMPLE_SERVER_NODEID
+
+// Transport Callbacks
+namespace {
+extern "C" void HandleDataModelMessage(const MessageHeader & header, System::PacketBuffer * buffer, SecureSessionMgrBase * mgr);
+
+class ServerCallback : public SecureSessionMgrCallback
+{
+public:
+    virtual void OnMessageReceived(const MessageHeader & header, Transport::PeerConnectionState * state,
+                                   System::PacketBuffer * buffer, SecureSessionMgrBase * mgr)
+    {
+        const size_t data_len = buffer->DataLength();
+        char src_addr[PeerAddress::kMaxToStringSize];
+
+        // as soon as a client connects, assume it is connected
+        VerifyOrExit(buffer != NULL, ChipLogProgress(App, "Received data but couldn't process it..."));
+        VerifyOrExit(header.GetSourceNodeId().HasValue(), ChipLogProgress(App, "Unknown source for received message"));
+
+        VerifyOrExit(state->GetPeerNodeId() != kUndefinedNodeId, ChipLogProgress(App, "Unknown source for received message"));
+
+        state->GetPeerAddress().ToString(src_addr, sizeof(src_addr));
+
+        ChipLogProgress(App, "Packet received from %s: %zu bytes", src_addr, static_cast<size_t>(data_len));
+
+        HandleDataModelMessage(header, buffer, mgr);
+        buffer = NULL;
+
+    exit:
+        // SendTo calls Free on the buffer without an AddRef, if SendTo was not called, free the buffer.
```

Comment:
This comment was wrong in `Server.cpp` and is wrong here.  There is no `SendTo` involved.  `HandleDataModelMessage` will free the buffer if it's called, and that should be documented in its API documentation.  This comment needs `s/SendTo/HandleDataModelMessage/g`.